File: client/src/types/world.ts
import Order from './order';
import Board, { compareBoards } from './board';
import Nation from './enums/nation';
import Location from './location';

type World = {
  iteration: number;
  boards: Board[];
  orders: Order[];
  winner: Nation | null;
};

export const findUnit = (world: World | null, location: Location) =>
  world?.boards.find((board) => compareBoards(board, location))?.units[location.region] ?? null;

export default World;

File: server/Models/World.cs
using Enums;

namespace Models;

public class World(int iteration, Board[] boards, Order[] orders, Nation? winner)
{
    public int Iteration { get; set; } = iteration;
    public Board[] Boards { get; set; } = boards;
    public Order[] Orders { get; set; } = orders;
    public Nation? Winner { get; set; } = winner;
}
